cmd/earthly/build_cmd.go

```diff
@@ -32,17 +45,11 @@ import (
 	"github.com/earthly/earthly/util/syncutil/semutil"
 	"github.com/earthly/earthly/util/termutil"
 	"github.com/earthly/earthly/variables"
-	"github.com/joho/godotenv"
-	"github.com/moby/buildkit/client/llb"
-	"github.com/moby/buildkit/session"
-	"github.com/moby/buildkit/session/auth"
-	dockerauthprovider "github.com/moby/buildkit/session/auth/authprovider"
-	"github.com/moby/buildkit/session/localhost/localhostprovider"
-	"github.com/moby/buildkit/session/socketforward/socketprovider"
-	"github.com/moby/buildkit/session/sshforward/sshprovider"
-	"github.com/moby/buildkit/util/entitlements"
-	"github.com/pkg/errors"
-	"github.com/urfave/cli/v2"
+)
+
+var (
+	orgMu     sync.Mutex
+	projectMu sync.Mutex
```

Comment:
we try to avoid globals as much as possible; I would mimic https://github.com/earthly/earthly/blob/main/analytics/counters.go and create `analytics/project_tracker.go`

```
type ProjectTracker struct {
        org string
        project string
	mutex  sync.Mutex
}

var projectTracker ProjectTracker

func (pt *ProjectTracker) AddProject(org, project string) {
     pt.Lock()
     defer pt.Unlock()
     if pt.org == "" {
        pt.org = org
        pt.project = project
    }

...ad similar method for getting
}
```

Reply:
I like this suggestion